### ISSUE TITLE
Permissions problem in files generated by sr_self_test 

### ISSUE THREAD

**Issue by toliver** [shadow-robot/sr-ros-interface#77]
From an email by Guillaume Walck:

Hi,

  context :
- In   R_groovy branch
- multi-user machine with permissions basically user rwx and group and
  others r-x
-  first user runs gazebo or edc_launch
  second user gets then errors messages + crashes of gazebo in sim or
  ethercat driver in real.

terminate called after throwing an instance of
'boost::filesystem3::filesystem_error'
what(): boost::filesystem::remove: Permission denied:
"/tmp/self_tests.bk/gazebo"

=> This comes from sr_self_test init that wants to re-use folders in
/tmp created with a different user so inappropriate rights for current
user to delete it

The strategy of keeping the self_tests logs sounds fine, but might I
suggest to use the $HOME/.ros/log folder to avoid sharing a folder in /tmp
Moreover, in standard usage (running the hand) the self_test does not
even use this folder. Why is the renaming stuff not done in the plot
test which requires this.

I checked, groovy comes with boost 1.46 but only 1.49 can set the
permissions.
 boost::filesystem::permissions(path_to_plots_,
boost::filesystem::owner_read
|

boost::filesystem::owner_write|

```
                             boost::filesystem::group_read
```

|
|

```
                             boost::filesystem::group_write
```

|
|

```
                             boost::filesystem::others_read
```

|
|

```
                             boost::filesystem::others_write);
```

We will currently deactivate the renaming in sr_self_test while waiting
for a fix.

**Comment by toliver** [comment]
From rhaschke@techfak.uni-bielefeld.de

I suggest, to simply use a random generated filename, e.g. using tmpnam:

http://www.cplusplus.com/reference/cstdio/tmpnam/

Best, Robert

**Comment by shadowmanos** [comment]
closed with #150
